internal/provider/provider.go

```diff
@@ -115,82 +144,90 @@ func New() *schema.Provider {
+						Type:        schema.TypeString,
+						Computed:    true,
+						Description: "Run this script on startup of an instance to initialize the agent.",
+					},
+					"arch": {
+						Type:         schema.TypeString,
+						ForceNew:     true,
+						Required:     true,
+						Description:  `The architecture the agent will run on. Must be one of: "amd64", "arm64".`,
+						ValidateFunc: validation.StringInSlice([]string{"amd64"}, false),
+					},
 					"auth": {
 						Type:         schema.TypeString,
 						Default:      "token",
+						ForceNew:     true,
 						Optional:     true,
 						Description:  `The authentication type the agent will use. Must be one of: "token", "google-instance-identity", "aws-instance-identity", "azure-instance-identity".`,
 						ValidateFunc: validation.StringInSlice([]string{"token", "google-instance-identity", "aws-instance-identity", "azure-instance-identity"}, false),
 					},
+					"env": {
+						ForceNew:    true,
+						Description: "A mapping of environment variables to set inside the workspace.",
+						Type:        schema.TypeMap,
+						Optional:    true,
+					},
 					"os": {
 						Type:         schema.TypeString,
+						ForceNew:     true,
 						Required:     true,
 						Description:  `The operating system the agent will run on. Must be one of: "linux", "darwin", or "windows".`,
 						ValidateFunc: validation.StringInSlice([]string{"linux", "darwin", "windows"}, false),
 					},
-					"arch": {
-						Type:         schema.TypeString,
-						Required:     true,
-						Description:  `The architecture the agent will run on. Must be one of: "amd64", "arm64".`,
-						ValidateFunc: validation.StringInSlice([]string{"amd64"}, false),
+					"startup_script": {
+						ForceNew:    true,
+						Description: "A script to run after the agent starts.",
+						Type:        schema.TypeString,
+						Optional:    true,
 					},
-					"value": {
-						Type:     schema.TypeString,
-						Computed: true,
+					"token": {
+						ForceNew:    true,
+						Description: `Set the environment variable "CODER_TOKEN" with this token to authenticate an agent.`,
+						Type:        schema.TypeString,
+						Computed:    true,
 					},
 				},
 			},
-		},
-		ResourcesMap: map[string]*schema.Resource{
-			"coder_agent": {
-				Description: "Use this resource to associate an agent.",
-				CreateContext: func(c context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
-					// This should be a real authentication token!
-					rd.SetId(uuid.NewString())
-					err := rd.Set("token", uuid.NewString())
-					if err != nil {
-						return diag.FromErr(err)
-					}
+			"coder_agent_instance": {
+				Description: "Use this resource to associate an instance ID with an agent for zero-trust " +
+					"authentication. This association is done automatically for \"google_compute_instance\", " +
+					"\"aws_instance\", \"azurerm_linux_virtual_machine\", and " +
+					"\"azurerm_windows_virtual_machine\" resources.",
```

Comment:
I find it slightly weird that we use the cloud names for aws and azure but not google. also, what does `azurerm` refer to?

Reply:
It's primarily because google doesn't in their resource naming, so I wanted to keep it consistent. `azurerm` stands for Azure Resource Manager- certainly _interestingly_ named...

Reply:
Makes sense!